File: Sources/Shared/Entity/Statics/EntityData.ts
export interface EntityCollision {
    /**
     * Fraction is the division factor of the size of the scaling that is pre-invoked when drawing. 1 means no scaling.
     */
    fraction: number;

    /**
     * The radius of circle.
     */
    radius: number;
}

/**
 * Minimum required data that all mobs/petals data must meet.
 */
export interface EntityData<I18n extends object> {
    /**
     * Internationalization of entity data.
     */
    i18n: Required<I18n>;

    /**
     * Collision data needed for collide.
     */
    collision: EntityCollision;
}

File: Sources/server/internal/native/entity_data.go
package native

type EntityCollision struct {
	// Fraction is the division factor of the size of the scaling that is pre-invoked when drawing. 1 means no scaling.
	Fraction float32 `json:"fraction"`

	// Radius is the radius of circle.
	Radius float32 `json:"radius"`
}

type EntityData[I18n any] struct {
	// I18n is internationalization of entity data.
	I18n I18n `json:"i18n"`

	// Collision is data needed for collide.
	Collision EntityCollision `json:"collision"`
}